src/__Plugin.ino

```diff
@@ -1200,14 +1200,46 @@ byte PluginCall(byte Function, struct EventStruct *event, String& str)
     case PLUGIN_WRITE:
     case PLUGIN_REQUEST:
     {
-      for (taskIndex_t task = 0; task < TASKS_MAX; task++)
+      taskIndex_t firstTask = 0;
+      taskIndex_t lastTask = TASKS_MAX;
+      String command = String(str);                         // Local copy to avoid warning in ExecuteCommand
+      int dotPos = command.indexOf(".");                    // Find first period
+      if (Function == PLUGIN_WRITE                          // Only applicable on PLUGIN_WRITE function
+        && dotPos > -1) {                                   // First precondition is just a quick check for a period (fail-fast strategy)
+        String thisTaskName = command.substring(1, dotPos); // Extract taskname
+        thisTaskName.replace("[", "");
+        thisTaskName.replace("]", "");
+        if (thisTaskName.length() > 0) {                    // Second precondition
+          taskIndex_t thisTask = findTaskIndexByName(thisTaskName);
+          if (thisTask != INVALID_TASK_INDEX) {             // Known taskname?
+#ifdef USES_P022                                            // Exclude P022 as it has rather explicit differences in commands when used with the [<TaskName>]. prefix
+            if (Settings.TaskDeviceEnabled[thisTask]        // and internally needs to know wether it was called with the taskname prefixed
+              && validPluginID_fullcheck(Settings.TaskDeviceNumber[thisTask])
+              && Settings.TaskDeviceDataFeed[thisTask] == 0) {
+              const deviceIndex_t DeviceIndex = getDeviceIndex_from_TaskIndex(thisTask);
+              if (validDeviceIndex(DeviceIndex) && Device[DeviceIndex].Number == PLUGIN_ID_022) {
+                thisTask = INVALID_TASK_INDEX;
+              }
+            }
+            if (thisTask != INVALID_TASK_INDEX) {
+#endif
+              firstTask = thisTask;
+              lastTask  = thisTask + 1;                     // Add 1 to satisfy the for condition
+              command   = command.substring(dotPos + 1);    // Remove [<TaskName>]. prefix
+#ifdef USES_P022
+            }
+#endif
+          }
+        }
+      }
+
+      for (taskIndex_t task = firstTask; task < lastTask; task++)
```

Comment:
The inner scope of this loop should be in another function.
In this loop you're trying to determine whether we need to "find" the task willing to accept the command (needs the for loop)
The added feature to directly address a task either by name or by index (not yet implemented) is to extract a specific task index.
So by having a separate function to perform this on a single task index you improve readability of the code.